monai/apps/datasets.py

```diff
@@ -352,63 +350,41 @@ class CVDecathlonDataset:
 
     def __init__(
         self,
-        root_dir: str,
-        task: str,
-        transform: Union[Sequence[Callable], Callable] = LoadNiftid(["image", "label"]),
-        download: bool = False,
+        dataset_cls,
+        nfolds: int = 5,
         seed: int = 0,
-        nsplits: int = 5,
-        cache_num: int = sys.maxsize,
-        cache_rate: float = 1.0,
-        num_workers: int = 0,
+        **dataset_params,
     ) -> None:
-        self.root_dir = root_dir
-        self.task = task
-        self.transform = transform
-        self.download = download
+        if not hasattr(dataset_cls, "_split_datalist"):
+            raise ValueError("dataset class must have _split_datalist API.")
+        self.dataset_cls = dataset_cls
+        self.nfolds = nfolds
         self.seed = seed
-        if nsplits < 2:
-            raise ValueError("nplits must be greater than 1 for cross validation.")
-        self.nsplits = nsplits
-        self.cache_num = cache_num
-        self.cache_rate = cache_rate
-        self.num_workers = num_workers
-
-    def get_dataset(self, fold: int, section: str):
+        self.dataset_params = dataset_params
+
+    def get_partition(self, fold: int, section: str):
         """
-        Generate dataset for specified fold index and section in the cross validation group.
+        Generate partition for specified fold index and section in the cross validation group.
 
         Args:
             fold: index of fold for training and validation.
             section: expected data section, can be: `training` or `validation`.
 
         """
-        if fold > self.nsplits - 1:
-            raise ValueError("fold index should be in [0, nsplit - 1].")
-        nsplits = self.nsplits
+        if fold > self.nfolds - 1:
+            raise ValueError("fold index should be in [0, nfold - 1].")
+        nfolds = self.nfolds
+        seed = self.seed
 
-        class _NsplitsDataset(DecathlonDataset):
+        class _NsplitsDataset(self.dataset_cls):  # type: ignore
             def _split_datalist(self, datalist: List[Dict]) -> List[Dict]:
                 if section not in ["training", "validation"]:
                     raise ValueError("section must be training or validation.")
-                length = len(datalist)
-                indices = np.arange(length)
-                self.randomize(indices)
-                fold_indices = split_dataset(nsplits, length)
+                data = partition_dataset(data=datalist, num_partitions=nfolds, shuffle=True, seed=seed)
+                train, val = generate_cross_validation_fold(partitions=data, fold_idx=fold)
```

Comment:
there are a few cross-validation types (https://en.wikipedia.org/wiki/Cross-validation_(statistics)), this is only implementing one of them. 

we need a more flexible API so that 
`cvdataset.get_partition(folds=[0, 2, 3])` will return a subset of the original data consisting of fold 0, 2 and 3. the user could use the subset in a training or validation phase, no need to specify it in `get_partition` interface

Reply:
should we specify the section (`training` or `validation`) if specify `folds=[0, 2, 3]`?

Reply:
no need to specify the section name

Reply:
but if only specify one fold index, will also specify section, right? Otherwise, people need to contact folds for training.

Reply:
Hi @wyli ,

I reviewed your request and I think it's not easy to achieve, because now we return dataset (subclass of `DecathlonDataset`) directly and it can apply **transforms** to the data. But if you want to return datalists of fold 0, 3, 4 instead of dataset, that's quite different, should we skip the transforms? I think we need a quick online discussion about it.

Thanks.

Reply:
it should be minor changes for the interface, previously was:
```py
cvdataset = CrossValidation(DecathlonDataset, nfolds=5, root_dir="./", seed=12345, download=True)
dataset_fold0_train = cvdataset.get_dataset(fold=0, section="training")
# execute training for fold 0 ...
```
more flexible version I'd like to have:
```py
cvdataset = CrossValidation(DecathlonDataset, nfolds=5, root_dir="./", seed=12345, download=True)
dataset_train = cvdataset.get_dataset(folds=[0, 1, 2])
dataset_val = cvdataset.get_dataset(folds=[3])
dataset_infer = cvdataset.get_dataset(folds=[4])
```

Reply:
OK, looks reasonable, I thought you want to get the datalists of fold 0, 3, 4 instead of constructing Datasets.
Will try to update it later.

Thanks.

Reply:
Hi @wyli ,

Updated according to your suggestion, could you please help review it again?
Thanks.